pallets/amm/src/lib.rs

```diff
@@ -399,13 +401,20 @@ decl_module! {
 
 			Self::deposit_event(RawEvent::RemoveLiquidity(who.clone(), asset_a, asset_b, liquidity_amount));
 
-			if liquidity_left == 0 {
+			ensure!(
+				u128::from(share_token) < MIN_POOL_LIQUIDITY_LIMIT,
+				Error::<T>::MinimalPoolLiquidityRequirementNotMet
+			);
+
+			if liquidity_left >= MIN_POOL_LIQUIDITY_LIMIT || liquidity_left.is_zero() {
```

Comment:
This doesn't make much sense to me. Why do you remove token if liquidity_left is more than limit?